src/Adapter/Carrier/CarrierDataProvider.php

```diff
@@ -55,6 +56,55 @@ public function getCarriers($id_lang, $active = false, $delete = false, $id_zone
         return Carrier::getCarriers($id_lang, $active, $delete, $id_zone, $ids_group, $modules_filters);
     }
 
+    /**
+     * Get all active carriers in given language, usable for choice form type
+     *
+     * @param int|null $languageId if not provided - will use the default language
+     *
+     * @return array carrier choices
+     */
+    public function getActiveCarriersChoices($languageId = null)
```

Comment:
How about inject configuration from adapter namespace in constructor of this class and rely on it instead?

Could be easier to unit test :)